sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs

```diff
@@ -286,3 +295,98 @@ pub fn secp256k1_verify(
 
     Ok(NativeResult::ok(cost, smallvec![Value::bool(result)]))
 }
+
+/***************************************************************************************************
+ * native fun secp256k1_sign (TEST ONLY)
+ * Implementation of the Move native function `secp256k1_sign(private_key: &vector<u8>, msg: &vector<u8>, hash: u8): vector<u8>`
+ * This function has two cost modes depending on the hash being set to`KECCAK256` or `SHA256`. The core formula is same but constants differ.
+ * If hash = 0, we use the `keccak256` cost constants, otherwise we use the `sha256` cost constants.
+ *   gas cost: 0 (because it is only for test purposes)
+ **************************************************************************************************/
+pub fn secp256k1_sign(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 3);
+
+    // The corresponding Move function, sui::ecdsa_k1::secp256k1_sign, is only used for testing, so
+    // we don't need to charge any gas.
+    let cost = 0.into();
```

Comment:
I've just hard-coded the zero gas cost here, but we could also get it from `NativesCostTable` if that's better?

Reply:
don't know, I think this is fine? Not sure if @tzakian has got opinion on the specifics here

Reply:
As long as the native is test only it should be fine :) And in fact is better for gas profiling. So LGTM.